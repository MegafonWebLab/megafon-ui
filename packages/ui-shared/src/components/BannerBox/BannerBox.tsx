import * as React from 'react';
import { cnCreate, Banner } from '@megafon/ui-core';

type BannerBoxProps = React.ComponentProps<typeof Banner>;

const cn = cnCreate('mfui-beta-banner-box');
const BannerBox: React.FC<BannerBoxProps> = ({ children, ...props }) => (
    <div className={cn()}>
        <Banner {...props}>{children}</Banner>
    </div>
);

export default BannerBox;
