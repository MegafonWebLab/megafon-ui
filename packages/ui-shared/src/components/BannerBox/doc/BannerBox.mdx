---
name: BannerBox
menu: Конструктор
route: components/bannerBox
---

import Example from "./BannerBox.example.mdx";

# BannerBox

Компонент **BannerBox** является оберткой над компонентом **Banner** из раздела **Компоненты** и имеет идентичные свойства и методы

<CopyToClipboardBox text="import { BannerBox } from '@megafon/ui-shared';" />

<Example />
