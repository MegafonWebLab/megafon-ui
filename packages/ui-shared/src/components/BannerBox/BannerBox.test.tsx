import * as React from 'react';
import { shallow } from 'enzyme';
import BannerBox from './BannerBox';

describe('<BannerBox />', () => {
    it('render BannerBox with default props', () => {
        const wrapper = shallow(
            <BannerBox>
                <div>test</div>
            </BannerBox>
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('render BannerBox with props', () => {
        const wrapper = shallow(
            <BannerBox
                loop
                autoPlay
                autoPlayDelay={1000}
                navTheme={'green'}
                onNextClick={jest.fn()}
                onPrevClick={jest.fn()}
                onDotClick={jest.fn()}
                onChange={jest.fn()}>
                <div>test</div>
            </BannerBox>
        );

        expect(wrapper).toMatchSnapshot();
    });
});
