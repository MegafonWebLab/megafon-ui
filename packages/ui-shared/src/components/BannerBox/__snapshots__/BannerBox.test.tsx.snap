// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BannerBox /> render BannerBox with default props 1`] = `
<div
  className="mfui-beta-banner-box"
>
  <Banner>
    <div>
      test
    </div>
  </Banner>
</div>
`;

exports[`<BannerBox /> render BannerBox with props 1`] = `
<div
  className="mfui-beta-banner-box"
>
  <Banner
    autoPlay={true}
    autoPlayDelay={1000}
    loop={true}
    navTheme="green"
    onChange={[MockFunction]}
    onDotClick={[MockFunction]}
    onNextClick={[MockFunction]}
    onPrevClick={[MockFunction]}
  >
    <div>
      test
    </div>
  </Banner>
</div>
`;
